src/preload-hover.js

```diff
@@ -25,14 +26,26 @@ export default class PreloadHover {
 
       uniqueLinks.forEach(link => {
         link.addEventListener('mouseover', () => {
-          timer = setTimeout(() =>{
-            const preload = document.createElement('link');
+          if(this.configuration.linkType === true) {
+            timer = setTimeout(() => {
+              const preload = document.createElement('link');
 
-            preload.setAttribute('rel', 'preload');
-            preload.setAttribute('href', link.href);
-
-            head.appendChild(preload);
+              preload.setAttribute('rel', 'preload');
+              preload.setAttribute('href', link.href);
+              head.appendChild(preload);
             }, this.configuration.debounceTime);
+          } else {
+            if(link.hostname != window.location.hostname) {
+              timer = setTimeout(() => {
+                const preload = document.createElement('link');
+
+                preload.setAttribute('rel', 'preload');
+                preload.setAttribute('href', link.href);
+
+                head.appendChild(preload);
```

Comment:
Yeah I can change the boolean type and refactor the code. I didn't mean to merge the code so I am going to move it back as I work on those.